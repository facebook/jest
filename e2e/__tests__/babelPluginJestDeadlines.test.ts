/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as path from 'path';
import {skipSuiteOnJasmine} from '@jest/test-utils';
import {extractSummary, runYarnInstall} from '../Utils';
import runJest from '../runJest';

const DIR = path.resolve(__dirname, '..', 'babel-plugin-jest-deadlines');

beforeAll(() => {
  runYarnInstall(DIR);
});

skipSuiteOnJasmine();

it('passes generally', () => {
  const result = runJest('babel-plugin-jest-deadlines', [
    '--no-cache',
    'plain.test.js',
  ]);
  expect(result.exitCode).toBe(1);
  expect(summaryWithoutTime(result)).toMatchSnapshot();
});

it('throws on deadline exceeded', () => {
  const result = runJest('babel-plugin-jest-deadlines', [
    '--no-cache',
    'typescript.test.ts',
  ]);
  expect(result.exitCode).toBe(1);
  expect(summaryWithoutTime(result)).toMatchSnapshot();
});

function summaryWithoutTime(result: {stderr: string}) {
  const summary = extractSummary(result.stderr);
  summary.rest = summary.rest.replace(
    /(waited here for) \d*\.?\d+ m?s\b/,
    '$1 <<REPLACED>>',
  );
  return summary;
}
