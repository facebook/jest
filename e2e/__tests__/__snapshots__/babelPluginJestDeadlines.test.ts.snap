// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes generally 1`] = `
Object {
  "rest": "FAIL __tests__/plain.test.js
  ● Test suite failed to run

    Cannot find module 'babel-plugin-jest-deadlines' from '/home/faux/clone/jest/e2e/babel-plugin-jest-deadlines'

      at Function.resolveSync [as sync] (../../node_modules/resolve/lib/sync.js:90:15)
          at Array.map (<anonymous>)",
  "summary": "Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /plain.test.js/i.",
}
`;

exports[`throws on deadline exceeded 1`] = `
Object {
  "rest": "FAIL __tests__/typescript.test.ts
  ● Test suite failed to run

    Cannot find module 'babel-plugin-jest-deadlines' from '/home/faux/clone/jest/e2e/babel-plugin-jest-deadlines'

      at Function.resolveSync [as sync] (../../node_modules/resolve/lib/sync.js:90:15)
          at Array.map (<anonymous>)",
  "summary": "Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites matching /typescript.test.ts/i.",
}
`;
